File: eknjiga_mobile_app/lib/models/author.dart
class Author {
  final int id;
  final String firstName;
  final String lastName;

  Author({required this.id, required this.firstName, required this.lastName});

  factory Author.fromJson(Map<String, dynamic> json) {
    return Author(
      id: json['id'],
      firstName: json['firstName'],
      lastName: json['lastName'],
    );
  }
}

File: eKnjiga/eKnjiga.Services/Database/Author.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace eKnjiga.Services.Database
{
    public class Author
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string LastName { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? BirthDate { get; set; }
        public DateTime? DeathDate { get; set; }

        [MaxLength(2000)]
        public string? Description { get; set; }
        
        public ICollection<BookAuthor> BookAuthors { get; set; } = new List<BookAuthor>();
    }
} 